smartsim/entity/model.py

```diff
@@ -128,21 +129,88 @@ def attach_generator_files(self, to_copy=None, to_symlink=None, to_configure=Non
         to_configure = init_default([], to_configure, (list, str))
         self.files = EntityFiles(to_configure, to_copy, to_symlink)
 
-    def colocate_db(
+    def colocate_db(self, *args, **kwargs):
+        warnings.warn(
+            (
+                "`colocate_db` has been deprecated and will be removed in a \n"
+                "future release. Please use `colocate_db_tcp` or `colocate_db_uds`."
+            ),
+            category=DeprecationWarning
+        )
+        self.colocate_db_tcp(*args, **kwargs)
+
+    def colocate_db_uds(
         self,
-        port=6379,
+        unix_socket="/tmp/redis.socket",
+        socket_permissions=755,
         db_cpus=1,
         limit_app_cpus=True,
+        debug=False,
+        **kwargs,
+    ):
+        """Colocate an Orchestrator instance with this Model over UDS.
+
+        This method will initialize settings which add an unsharded
+        database to this Model instance. Only this Model will be able to communicate
+        with this colocated database by using Unix Domain sockets.
+
+        Extra parameters for the db can be passed through kwargs. This includes
+        many performance, caching and inference settings.
+
+        .. highlight:: python
+        .. code-block:: python
+
+            ex. kwargs = {
+                maxclients: 100000,
```

Comment:
Can we change the dict name and make the keys strings to make the code block valid python? (e.g. `ex. kwargs -> ex_kwargs` and `maxclients -> "maxclients"`)

Reply:
How dare. Valid python? You think we're professionals over here?!